src/lwc/bdiBatchNumberSettings/bdiBatchNumberSettings.js

```diff
@@ -171,29 +173,50 @@ export default class bdiBatchNumberSettings extends LightningElement {
         }
     }
 
-    handleCreateBatchNumber() {
-        const fields = {};
-        fields[Object_API_Name.fieldApiName] = DataImportBatch.objectApiName;
-        fields[Field_API_Name.fieldApiName] = Batch_Number.fieldApiName;
-        fields[Starting_Number.fieldApiName] = this.startingNumber;
-        fields[Display_Format.fieldApiName] = this.displayFormat;
-        fields[Description.fieldApiName] = this.description;
+    async handleSave() {
+        await this.save();
+        this.reset();
+        this.fetchAutoNumbers();
+    }
 
-        const record = {
-            fields: fields
+    async save() {
+        this.isLoading = true;
+        try {
+            const autoNumber = await save({autoNumber: JSON.stringify({fields: this.fields})});
+            return autoNumber;
+        } catch (err) {
+            this.error = err;
+            this.displayErrorOnInputs(this.errorDetails);
+        } finally {
+            this.isLoading = false;
         }
+    }
+
+    async handleSaveAndActivate() {
+        const autoNumber = await this.save();
+        await this.activate(autoNumber.Id);
+        this.fetchAutoNumbers();
+    }
 
+    async activate(autoNumberId) {
         this.isLoading = true;
-        save({autoNumber: JSON.stringify(record)})
-            .then(() => {
-                this.reset();
-                this.fetchAutoNumbers();
-            })
-            .catch(error => {
-                this.error = error;
-                this.displayErrorOnInputs(this.errorDetails);
-            })
-            .finally(() => this.isLoading = false);
+        try {
+            await activate({autoNumberId: autoNumberId});
+        } catch (err) {
+            this.error = err;
+        } finally {
+            this.isLoading = false;
+        }
+    }
+
+    get fields() {
```

Comment:
I broke this out into its own property, and changed save and activate to be async functions for better readability.